tools/gcs-bucket-mover/gcs_bucket_mover/bucket_mover_service.py

```diff
@@ -85,40 +107,101 @@ def move_bucket(conf):
 
-    with yaspin(text='Re-create source bucket in target project') as spinner:
+    spinner_text = 'Re-creating source bucket in target project'
+    _LOGGER.log_text(spinner_text)
+    with yaspin(text=spinner_text) as spinner:
         _create_bucket(spinner, config, config.target_storage_client,
                        config.bucket_name, source_bucket_details)
         spinner.ok(_CHECKMARK)
 
-    with yaspin(text='Assign STS permissions to new source bucket') as spinner:
+    spinner_text = 'Assigning STS permissions to new source bucket'
+    _LOGGER.log_text(spinner_text)
+    with yaspin(text=spinner_text) as spinner:
         _assign_sts_iam_roles(sts_account_email, config.target_storage_client,
                               config.target_project, config.bucket_name, False)
         spinner.ok(_CHECKMARK)
 
     _run_and_wait_for_sts_job(sts_client, config.target_project,
                               config.temp_bucket_name, config.bucket_name)
 
-    with yaspin(text='Delete empty temp bucket') as spinner:
+    spinner_text = 'Deleting empty temp bucket'
+    _LOGGER.log_text(spinner_text)
+    with yaspin(text=spinner_text) as spinner:
         target_temp_bucket.delete()
         spinner.ok(_CHECKMARK)
 
-    spinner_text = 'Remove STS permissions from new source bucket'
+    spinner_text = 'Removing STS permissions from new source bucket'
+    _LOGGER.log_text(spinner_text)
     with yaspin(text=spinner_text) as spinner:
         _remove_sts_iam_roles(sts_account_email, config.target_storage_client,
                               config.bucket_name)
         spinner.ok(_CHECKMARK)
 
+    _LOGGER.log_text('Completed GCS Bucket Mover')
+
+
+def _lock_down_bucket(spinner, bucket, lock_file_name, service_account_email):
+    """Change the ACL/IAM on the bucket so that only the service account can access it.
+
+    Args:
+        spinner: The spinner displayed in the console
+        bucket: The bucket object to lock down
+        lock_file_name: The name of the lock file
+        service_account_email: The email of the service account
+    """
+
+    if _does_lock_file_exist(bucket, lock_file_name):
+        spinner.fail('X')
+        msg = 'The lock file exists in the source bucket, so we cannot continue'
+        _LOGGER.log_text(msg)
+        raise SystemExit(msg)
+
+    spinner.ok(_CHECKMARK)
+    msg = 'Locking down the bucket by revoking all ACLs/IAM policies'
+    spinner.text = msg
+    _LOGGER.log_text(msg)
+
+    # Turn off any bucket ACLs
+    bucket.acl.save_predefined('private')
+
+    # Revoke all IAM access and then add the service account as an admin
+    account = 'serviceAccount:' + service_account_email
+    policy = bucket.get_iam_policy()
+    for role in policy.keys():
+        for member in policy[role].copy():
+            policy[role].discard(member)
+
+    policy['roles/storage.admin'].add(account)
+    bucket.set_iam_policy(policy)
+
+
+def _does_lock_file_exist(bucket, lock_file_name):
```

Comment:
I don't think this function should exist. It doesn't add any value to the caller at all, and the code would be shorter if you just called `storage.Blob(lock_file_name, bucket).exists()` instead of calling this function.